### PR TITLE
add workflow to periodically cancel stale workflows

### PR DESCRIPTION
**Pull request by m-vdb**
**Proposed changes**:
- Add workflow to periodically cancel stale workflows

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: .github/workflows/cancel-stale-workflows.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+jobs:
+  cancel-runs:
+    # Prevent forks from running this to be nice
+    if: github.repository == 'RasaHQ/rasa'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: n1hility/cancel-previous-runs@v2
+      with:
+        token: ${{ secrets.RASABOT_GITHUB_TOKEN }}
+        workflow: 351973
```
Comment (m-vdb) [inline .github/workflows/cancel-stale-workflows.yml:16]: checked on API

File: .github/workflows/cancel-stale-workflows.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    - name: Notify slack on failure
+      if: failure() && github.event_name == 'schedule'
+      env:
+        SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      uses: voxmedia/github-action-slack-notify-build@v1
+      with:
+        channel_id: ${{ secrets.SLACK_ALERTS_CHANNEL_ID }}
```
Comment (m-vdb) [inline .github/workflows/cancel-stale-workflows.yml:24]: all secrets are already present

Comment (tmbo) [reply .github/workflows/cancel-stale-workflows.yml:24]: yes using the rasabot account, I've created a new token and added it

**Review by tmbo — APPROVED** [review]
looks good - is there a cancel job as part of the ci workflow that should be removed?

**Comment by m-vdb** [comment]
@tmbo no there wasn't

**Pull request merged**